mobile/src/bundles/talent/store/slice.ts

```diff
@@ -2,52 +2,63 @@ import { createSlice } from '@reduxjs/toolkit';
 
 import { DataStatus } from '~/bundles/common/enums/enums';
 import { type ValueOf } from '~/bundles/common/types/types';
-import { type ProfileStepDto } from '~/bundles/talent/types/types';
+import { type UserDetailsResponseDto } from '~/bundles/talent/types/types';
 
-import { completeProfileStep } from './actions';
+import {
+    createTalentDetails,
+    getTalentDetails,
+    setCompletedStep,
+    updateOnboardingData,
+} from './actions';
 
 type State = {
     dataStatus: ValueOf<typeof DataStatus>;
-    profileStepData: ProfileStepDto | null;
+    onboardingData: UserDetailsResponseDto | null;
+    completedStep: number;
 };
 
 const initialState: State = {
     dataStatus: DataStatus.IDLE,
-    profileStepData: null,
+    onboardingData: null,
+    completedStep: 0,
 };
 
 const { reducer, actions, name } = createSlice({
     initialState,
     name: 'talents',
     reducers: {},
     extraReducers(builder) {
-        builder.addCase(completeProfileStep.fulfilled, (state, action) => {
-            const {
-                profileName,
-                salaryExpectation,
-                employmentType,
-                experienceYears,
-                jobTitle,
-                location,
-                description,
-            } = action.payload;
-            state.dataStatus = DataStatus.FULFILLED;
-            state.profileStepData = {
-                profileName,
-                salaryExpectation,
-                employmentType,
-                experienceYears,
-                jobTitle,
-                location,
-                description,
-            };
-        });
-        builder.addCase(completeProfileStep.pending, (state) => {
-            state.dataStatus = DataStatus.PENDING;
-        });
-        builder.addCase(completeProfileStep.rejected, (state) => {
-            state.dataStatus = DataStatus.REJECTED;
+        builder.addCase(setCompletedStep, (state, action) => {
+            state.completedStep = action.payload;
         });
+        builder.addMatcher(
+            (action) =>
+                action.type === createTalentDetails.fulfilled.type ||
+                action.type === updateOnboardingData.fulfilled.type ||
+                action.type === getTalentDetails.fulfilled.type,
+            (state, action) => {
+                state.dataStatus = DataStatus.FULFILLED;
+                state.onboardingData = action.payload;
```

Comment:
Do all these actions return data related to all steps? If yes, then okay, if only patched data, then you should write like this: state.onboardingData = {...state.onboardingData, ...action.payload};

Reply:
According to Swagger - yes, all queries are expected to return the full object with user-details.